### PR TITLE
Feat/mark enhancement

### LINKED ISSUE #509: [Feature] new types of makerLine
### What problem does this feature solve?

* 复合增长箭头

image

* 总计差异箭头
image

* 层级差异箭头
image

### What does the proposed API look like?

need to design

### LINKED ISSUE #699: [Feature] markLine label auto position
### What problem does this feature solve?

when markline.autoRange=true, label of the markline will be clipped:
image

the position of the label should be auto changed to avoid clip like this:
image

It will first use the position in label.position if it can be fully displayed. 

### What does the proposed API look like?

markline:{
autoRange:true,
label:{
autoPosition:true
}

}

### PR DESCRIPTION
**Pull request by kkxxkk2019**
1. markLine 和 markArea 的 label 新增 `confine` 属性用于将文本自动调整到显示范围内
2. 新增 `type: 'type-step'` 类型 markLine 标注
image

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

closed #699 
closed #509 
### 🔗 Related PR link

https://github.com/VisActor/VRender/pull/472

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/marker/mark-area/mark-area.ts [hunk 1]
```diff
@@ -112,29 +115,29 @@ export class MarkArea extends BaseMarker<IMarkAreaSpec & IMarkAreaTheme> impleme
...
-      ]);
-      clipRange = {
-        x: minX,
-        y: minY,
-        width: maxX - minX,
-        height: maxY - minY
-      };
-    }
-    this._markerComponent?.setAttributes({
+
```
Comment (skie1997) [inline packages/vchart/src/component/marker/mark-area/mark-area.ts:118]: 如果confine: false, autoInRange: false, limitRect还需要计算吗？

File: packages/vchart/src/component/marker/mark-line/interface/spec.ts [hunk 2]
```diff
@@ -6,23 +6,28 @@ import type {
...
+      (
+        | IMarkLineXSpec // 标注目标：笛卡尔坐标系坐标空间
+        | IMarkLineYSpec // 标注目标：笛卡尔坐标系坐标空间
+        // | IMarkLineAngleSpec // TODO: 标注目标：极坐标系坐标空间
+        // | IMarkLineRadiusSpec // TODO: 标注目标：极坐标系坐标空间
+        | IMarkLineCoordinateSpec // 标注目标：数据元素
+        | IMarkLinePositionsSpec
+      ) &
+      IMarkLineTheme)
+  | IStepPMarkLineSpec; // 标注目标：任意位置
```
Comment (skie1997) [inline packages/vchart/src/component/marker/mark-line/interface/spec.ts:30]: 这里的拼写是不是`IStepMarkLineSpec`？

Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-line/interface/spec.ts:30]: O(∩_∩)O哈哈~，是的

File: packages/vchart/src/component/marker/mark-area/mark-area.ts (hunk 1 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-area/mark-area.ts:118]: done

**Review by skie1997 — APPROVED** [review]

**Pull request merged**